cpp/src/arrow/util/decimal-test.cc

```diff
@@ -482,6 +482,49 @@ TEST(Decimal128Test, Multiply) {
   ASSERT_EQ(result.ToIntegerString(), "60501");
 }
 
+TEST(Decimal128Test, Divide) {
+  Decimal128 result;
```

Comment:
This is valid for the next 3 tests:
- Why the usage of a temporary variables, e.g. `ASSERT_EQ(Decimal128("20100") / Decimal128("301"), ...)`?
- Why the usage of `ToIntegerString`, just compare `Decimal("66")`?
- Would it be better to explicit some int64/int128 constructors such that you can at least call with `Decimal(66)`?

The end would look like
```
  ASSERT_EQ(Decimal(20100) / Decimal128(301), Decimal128(66));
```
>

- Please do some _minimal_ property testing against known types, e.g.

```
for (x: rand_int32_range())
  for (y: rand_int32_range())
    ASSERT_EQ(Decimal(x) / Decimal128(y), Decimal128(x/y));
```
- I usually try to test all obvious edge cases and some close bounds, e.g.
`{INT_MIN, 0, INT_MAX} x {-2,-1,0,1,2}`

Reply:
The int64/int32 constructors already exist. I've added tests and fixed as suggested.